packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
+                        actionsSection: _buildActions(),
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            );
+          },
         ),
       ),
     );
   }
 }
 
-/// A button typically used in a [CupertinoAlertDialog].
-///
-/// See also:
-///
-///  * [CupertinoAlertDialog], a dialog that informs the user about situations
-///    that require acknowledgement
-class CupertinoDialogAction extends StatelessWidget {
-  /// Creates an action for an iOS-style dialog.
-  const CupertinoDialogAction({
-    this.onPressed,
-    this.isDefaultAction = false,
-    this.isDestructiveAction = false,
-    @required this.child,
-  }) : assert(child != null);
+// Reports whether or not iOS is currently in accessibility mode.
+class _CupertinoAccessibility extends InheritedWidget {
+  const _CupertinoAccessibility({
+    @required double textScaleFactor,
+    Widget child,
+  }) : isInAccessibilityMode = textScaleFactor > _kMaxRegularTextScaleFactor,
+        super(child: child);
 
-  /// The callback that is called when the button is tapped or otherwise
-  /// activated.
-  ///
-  /// If this is set to null, the button will be disabled.
-  final VoidCallback onPressed;
+  final bool isInAccessibilityMode;
 
-  /// Set to true if button is the default choice in the dialog.
-  ///
-  /// Default buttons are bold.
-  final bool isDefaultAction;
+  static _CupertinoAccessibility of(BuildContext context) {
+    return context.inheritFromWidgetOfExactType(_CupertinoAccessibility);
+  }
 
-  /// Whether this action destroys an object.
-  ///
-  /// For example, an action that deletes an email is destructive.
-  final bool isDestructiveAction;
+  @override
+  bool updateShouldNotify(_CupertinoAccessibility oldWidget) {
+    return isInAccessibilityMode != oldWidget.isInAccessibilityMode;
+  }
+}
 
-  /// The widget below this widget in the tree.
-  ///
-  /// Typically a [Text] widget.
-  final Widget child;
+// iOS style layout policy widget for sizing an alert dialog's content section and
+// action button section.
+//
+// See [_RenderCupertinoDialog] for specific layout policy details.
+class _CupertinoDialogRenderWidget extends RenderObjectWidget {
+  const _CupertinoDialogRenderWidget({
+    Key key,
+    @required this.contentSection,
+    @required this.actionsSection,
+  }) : super(key: key);
 
-  /// Whether the button is enabled or disabled. Buttons are disabled by
-  /// default. To enable a button, set its [onPressed] property to a non-null
-  /// value.
-  bool get enabled => onPressed != null;
+  final Widget contentSection;
+  final Widget actionsSection;
 
   @override
-  Widget build(BuildContext context) {
-    TextStyle style = _kCupertinoDialogActionStyle;
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoDialog(
+      dividerThickness: _kDividerThickness / MediaQuery.of(context).devicePixelRatio,
```

Comment:
There wouldn't be any point because, as you can see from the referenced line, that divider thickness can't change.  It's being set by a constant divided by the device's pixel density.  If it were variable, then yes, we'd definitely want the ability to mutate it.  But I don't see the point in this case.